launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -488,11 +549,22 @@ public Process launch() throws IOException {
    * In all cases, the logger name will start with "org.apache.spark.launcher.app", to fit more
    * easily into the configuration of commonly-used logging systems.
    *
+   * If the application is launched as a thread, the log redirection methods are not supported,
+   * and the parent process's output and log configuration will be used.
+   *
    * @since 1.6.0
    * @param listeners Listeners to add to the handle before the app is launched.
    * @return A handle for the launched application.
    */
   public SparkAppHandle startApplication(SparkAppHandle.Listener... listeners) throws IOException {
+    if (launchAsThread) {
+      return startApplicationAsThread(listeners);
```

Comment:
Add a `checkArgument` call to make sure `builder.childEnv` is empty in this case, since env variables are not supported in thread mode.